### PR TITLE
Closes #828: [Pocket] Introduce pocket mega tile on first launch

### LINKED ISSUE #828: [Pocket] Introduce pocket mega tile on first launch
Why:

- Explain why FFTV users are seeing this new tile in their homescreen

What:

- An overlay that displays at first launch with the pocket mega tile and, a title and short explanation that give more context about the feature
- A link to a web page that gives further information about the feature

Who:

- All FFTV users

Acceptance Criteria:

- Users can dismiss the overlay and access the homescreen
- Users can find out more via the "Learn more" link

Metrics/KPIs:

- Click on "Learn more"
- Click on pocket mega tile right after the overlay is dismissed

### PR DESCRIPTION
**Pull request by bkmunar**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -162,6 +164,19 @@ class BrowserFragment : IWebViewLifecycleFragment() {
...
+            PreferenceManager.getDefaultSharedPreferences(context)
+                    .edit()
+                    .putBoolean(POCKET_ONBOARD_SHOWN_PREF, true)
+                    .apply()
+            return true
+        }
+        return false
+    }
+
+
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/browser/BrowserFragment.kt]: \[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1

File: app/src/main/java/org/mozilla/focus/home/pocket/PocketOnboardingFragment.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
+            inflater.inflate(R.layout.fragment_pocket_onboarding, container, false)
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        pocket_onboarding_button.setOnClickListener {
+            ScreenController.showPocketScreen(fragmentManager)
+        }
+    }
+
+
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/home/pocket/PocketOnboardingFragment.kt]: \[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1

**Comment by liuche** [comment]
@bkmunar you mentioned that you were going to use Activity instead of AppCompatActivity - did you try setting the theme with `android:windowIsTranslucent`, `android:windowBackground` to transparent, and `android:windowContentOverlay` to `null` like this [stackoverflow](https://stackoverflow.com/questions/2176922/how-do-i-create-a-transparent-activity-on-android) answer?

We should use AppCompatActivity if at all possible, because it allows compatibility with support libraries, whereas using just Activity might break things on certain devices/versions, and might not work with future features.

@bkmunar Also, do you have a WIP with everything except the transparency done? I see a bunch of commits in this PR, but I'm confused by them and they aren't organized. Is everything done, except the transparency? In the future, if it's just one small thing that can be separated out, you should do everything else in clean commits, and then do the last part as a final commit, rather than lots of WIP commits that touch everything while trying to figure it out.

**Comment by bkmunar** [comment]
The reason I have all those commits was I was kind of in a sticky thought process so everything was touching everything at one point. I'm planning on recommitting them once all my thoughts are separated since it was a difficult one for me to sort out.

**Comment by bkmunar** [comment]
We are still waiting on an asset for the help button in the pocket video feed. Other than that, I think it is up to date with the mock

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -15,6 +15,14 @@
         <item name="android:statusBarColor">@android:color/transparent</item>
     </style>
 
+    <style name="AppTheme.Transparent" parent="Theme.AppCompat.NoActionBar">
+        <item name="android:windowNoTitle">true</item>
+        <item name="android:windowIsTranslucent">true</item>
+        <item name="android:windowBackground">@android:color/transparent</item>
+        <item name="android:statusBarColor">@android:color/transparent</item>
```
[the remark this answers is not in the document]
Comment (bkmunar) [reply app/src/main/res/values/styles.xml]: I confirmed that only windowIsTranslucent and windowBackground is needed to be set for the transparency to work

File: app/src/main/java/org/mozilla/focus/utils/Settings.kt [hunk 4]
```diff
@@ -40,6 +40,9 @@ class Settings private constructor(context: Context) {
     fun shouldShowOnboarding(): Boolean =
```
Comment (liuche) [inline app/src/main/java/org/mozilla/focus/utils/Settings.kt]: Nit: rename this to `shouldShowTurboModeOnboarding`

**Pull request merged**